var/spack/repos/builtin/packages/gobject-introspection/package.py

```diff
@@ -46,6 +47,9 @@ class GobjectIntrospection(Package):
     depends_on("flex", type="build")
     depends_on("pkg-config@0.9.0:", type="build")
 
+    # GobjectIntrospection does not build with sed from darwin:
+    depends_on('sed', when=sys.platform == 'darwin', type='build')
```

Comment:
You could also use `when='platform=darwin'` and not have to import anything. It would be good to start using Spack's platform identifiers more.